StitchCore/StitchCore/Source/Core/Storage.swift

```diff
@@ -0,0 +1,132 @@
+import Foundation
+
+private let latestVersion = 1
+private let versionKey = "__stitch_storage_version__"
+
+/// Run a migration on the currently used storage
+/// that checks to see if the current version is up to date.
+/// If the version has not been set, this method will migrate
+/// to the latest version.
+/// - parameter suiteName: namespace to key with
+/// - parameter storage: storage to check for version key
+internal func runMigration(suiteName: String, storage: inout Storage) {
+    switch storage.value(forKey: versionKey) {
+    case .none:
+    #if !os(Linux)
+        // map each of the store's keys to a Promise
+        // that fetches the each value for each key,
+        // sets the old value to the new "namespaced" key
+        // remove the old key value pair,
+        // and set the version number
+        // we only care about UserDefaults here since no other
+        // `Storage` type was available before version 1
+        guard var originalStorage = UserDefaults.init(suiteName: "com.mongodb.stitch.sdk.UserDefaults") else {
```

Comment:
Technically, it will always "run". But once the value is set, next time this block is hit, the version `case` will be the `default` case, which is empty. If it would be more legible, we could add a simple boolean check to see if `storage.value(forKey: versionKey)` is equal to `latestVersion`.

Reply:
oh I missed that. fine as is!